### PR TITLE
#21811 - Rest-API updating product stock_item deletes downloadable_product_links

### PR DESCRIPTION
**Pull request by sanjay-wagento**
### Fixed Issues (if relevant)

1. magento/magento2#21811: Rest-API updating product stock_item deletes downloadable_product_links

### Manual testing scenarios (*)

1. Create regular downloadable product, upload downloadable content
2. Update product stock_item values using a PUT call. For example:
Rest Url: DOMAIN_URL/rest/V1/products/240-LV04
Method: PUT
Body: `"product": {"sku": "TESTITEM", "status": "1", "name": "Test Name", "price": "11.99", "type_id": "downloadable", "extension_attributes": {"stock_item": {"manage_stock": false}}}}`

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by sanjay-wagento** [comment]
@magento-team , I have reviewed error but I do not understand. Can you please tell me what is the issue?

**Comment by slavvka** [comment]
@magento run all tests

File: app/code/Magento/Downloadable/Model/Link/UpdateHandler.php [hunk 1]
```diff
@@ -44,16 +44,18 @@ public function execute($entity, $arguments = [])
         $links = $entity->getExtensionAttributes()->getDownloadableProductLinks() ?: [];
         $updatedLinks = [];
         $oldLinks = $this->linkRepository->getList($entity->getSku());
-        foreach ($links as $link) {
-            if ($link->getId()) {
-                $updatedLinks[$link->getId()] = true;
+        if (!empty($links)) {
```
Comment (slavvka) [inline app/code/Magento/Downloadable/Model/Link/UpdateHandler.php:47]: It makes sense to move this condition one line upper to not get list of existing links

File: app/code/Magento/Downloadable/Model/Sample/UpdateHandler.php [hunk 2]
```diff
@@ -44,16 +44,18 @@ public function execute($entity, $arguments = [])
         $samples = $entity->getExtensionAttributes()->getDownloadableProductSamples() ?: [];
         $updatedSamples = [];
         $oldSamples = $this->sampleRepository->getList($entity->getSku());
-        foreach ($samples as $sample) {
-            if ($sample->getId()) {
-                $updatedSamples[$sample->getId()] = true;
+        if (!empty($samples)) {
```
Comment (slavvka) [inline app/code/Magento/Downloadable/Model/Sample/UpdateHandler.php:47]: The same

**Review by slavvka — CHANGES_REQUESTED** [review]
Hey @sanjay-wagento, thank you for the fix. Please make changes according to my comment

**Comment by slavvka** [comment]
And let's wait for the builds

**Comment by slavvka** [comment]
Hey @sanjay-wagento please look at the static tests, they are failing. Please ignore Database Compare tests for now. Also we need to have these changes covered with automated tests. I think WebApi is the most suitable

**Comment by slavvka** [comment]
@magento run Database Compare build

**Comment by sidolov** [comment]
@sanjay-wagento , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**